rfcs/rfc-create-ui-architecture.md

```diff
@@ -0,0 +1,146 @@
+  - see [existing UI address schema](https://github.com/bcgov/bcrs-business-create-ui/blob/master/src/schemas/office-address-schema.ts)
+- other validation rules or functions (eg, for Vuetify components)
+- doesn't have to be one large file -- could import/specify references to other objects
+- hopefully make this clear/clean enough that a Business Analyst could (help) verify the config per filing type
+- probably don't need to fetch this resources/config object from server/db at this time
+  - we don't need the flexibility at the moment (and our code may not support the flexibility anyway)
+  - resources/config object can just be part of the codebase for now (can be changed later if needed)
+
+### 2. Code Changes
+
+As new filing types are added, update the code to use the resources for the current filing type, eg:
+
+- conditional logic based on flags/rules
+- display specific titles/text
+- do not use `if (a) then ... else ...` as the else (fallback) case will make it harder in the future to understand when rules apply or don't apply
+- do use `if (a) then ... else if (b) then ...` -- a condition for each filing type
+- avoid cascading rules; instead, use more specific conditions (*)
+- add new resources to all existing filing types
+  - if a rule doesn't apply, add an explicit code comment so it doesn't look like an oversight
+- refactor existing resources as needed to maintain clarity/cleanliness
+
+(*) Although default/overarching rules, with specific exceptions for some filing types, initially sound like a good idea, their use should be minimized or heavily scrutinized, as they can lead to complicated logic -- which will lead to complicated or hard-to-understand (and therefore, hard to implement, debug and maintain) code.
+
+## Principles
+
+- start small and add to it
+- keep a sane architecture as we build upon it
+- keep it general so that it can be extended easily
+- refactor resources/config object as needed
+- minimize code refactoring
+- create unit tests to ensure resource object and components continue to function through code changes
+- parameterize unit tests to minimize duplication
+- create and use common or [shared components](https://github.com/bcgov/bcrs-shared-components) as opportunities arise
+
+## Implementation
+
+### Option 1
+
+- implement/enhance a component once with all possible permutations to handle all different filings
+- difficult or impossible to build correctly (we can't know everything up front)
+- not possible to test end-to-end during development
+- cannot release (deploy) new filing types one by one
+
+### Option 2
+
+- add filings types one by one
+- enhance components as we go, adding in per-type handling where needed
+- this is likely easier and more flexible as we only update what we need, when we need it (ie, Agile)
+- this is probably less work, even though we'll touch components often
+- can release new filing types as they are implemented
+
```

Comment:
### Option 3

A blend of options 1 & 2.  
- add filing types in small bundles ie. 1 bundle IA - BC, CCC, ULC, 1 bundle Continuation Ins - BEN, BC, CCC, ULC
- have awareness of other permutations  outside of bundles to inform how configurations in bundle
- not possible to test end-to-end developments until bundle completed
- can still enhance components as we go, adding in per-type handling where needed
- still only update what we need, when we need it, but with awareness when designing configs should make it easier to update in future bundles.
- can release new filing types as they are implemented

Reply:
Updated.